src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt

```diff
@@ -61,45 +57,51 @@ class HistoryService(private val project: Project) :
 
   @Synchronized
   fun updateDefaultContextState(contextState: EnhancedContextState) {
-    state.defaultEnhancedContext = EnhancedContextState()
-    state.defaultEnhancedContext?.copyFrom(contextState)
+    getOrCreateActiveAccountHistory().defaultEnhancedContext = EnhancedContextState()
+    getOrCreateActiveAccountHistory().defaultEnhancedContext?.copyFrom(contextState)
   }
 
   @Synchronized
   fun getContextReadOnly(internalId: String): EnhancedContextState? {
     return copyEnhancedContextState(
-        state.chats.find { it.internalId == internalId }?.enhancedContext)
+        getOrCreateActiveAccountHistory()
+            .chats
+            .find { it.internalId == internalId }
+            ?.enhancedContext)
   }
 
   @Synchronized
   fun getDefaultContextReadOnly(): EnhancedContextState? {
-    return copyEnhancedContextState(state.defaultEnhancedContext)
+    return copyEnhancedContextState(getOrCreateActiveAccountHistory().defaultEnhancedContext)
   }
 
   @Synchronized
-  fun getDefaultLlm(): LLMState? {
-    val defaultLlm = LLMState()
-    defaultLlm.copyFrom(state.defaultLlm ?: return null)
-    return defaultLlm
-  }
+  fun getDefaultLlm(): LLMState? =
+      CodyAuthenticationManager.getInstance(project).getActiveAccount()?.let { account ->
+        findHistory(account.id)?.defaultLlm?.let { llm -> LLMState().also { it.copyFrom(llm) } }
+      }
```

Comment:
a lot of nesting here 😕 can you make it more readable?

Reply:
Done